packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -402,6 +401,8 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       });
 
       if (context.featureFlags.getBoolean('useSubUsernameForDefaultIdentityClaim')) {
+        const roleDefinitions = acm.getRoles().map(role => this.roleMap.get(role)!);
```

Comment:
@alharris-at the code is now getting all the roles instead of only the `readRoles` (see line 352). (variable moving is unrelated) We found a bug that only affected mutation operations because this method was only applied to read/query roles.

Reply:
ahh yes, it turns out I AM blind. thanks for clarifying.